src/hub/ShareClassManager.sol

```diff
@@ -519,13 +521,26 @@ contract ShareClassManager is Auth, IShareClassManager {
         uint128 newIssuance = metrics[scId_].totalIssuance - amount;
         metrics[scId_].totalIssuance = newIssuance;
 
-        emit RevokeShares(poolId, scId_, epochId[poolId], navPerShare, navPerShare.mulUint128(newIssuance), amount, 0);
+        emit RevokeShares(poolId, scId_, epochId[poolId], navPerShare.mulUint128(newIssuance), navPerShare, newIssuance, amount, 0);
     }
 
     /// @inheritdoc IShareClassManager
-    function updateShareClassNav(PoolId poolId, ShareClassId scId_) external view auth returns (uint128, D18) {
+    function updateShareClass(PoolId poolId, ShareClassId scId_, D18 navPerShare, bytes calldata data) external auth returns (uint128, D18) {
         require(exists(poolId, scId_), ShareClassNotFound());
-        revert("unsupported");
+
+        ShareClassMetrics storage m = metrics[scId_];
+        m.navPerShare = navPerShare;
+        emit UpdateShareClass(poolId, scId_, navPerShare.mulUint128(m.totalIssuance), navPerShare, m.totalIssuance, data);
```

Comment:
NIT. Do we really need `navPerShare.mulUint128(m.totalIssuance)` if both factors are already in the event?

Reply:
I suppose we do. @mustermeiszer added it. AFAIK because it's not easy for Apps to do that calculation. Does that still hold true?